File: app/lib/models/prestador.dart
import 'avaliacao.dart';

/// Um prestador de serviço cadastrado no app.
class Prestador {
  final String id;
  final String nome;
  final String categoriaId;
  final String especialidade; // ex: "Instalações elétricas residenciais"
  final String cidade;
  final String sobre; // descrição livre do profissional
  final String? fotoUrl; // foto de perfil (null = usa avatar com inicial)
  final bool verificado; // selo de identidade verificada

  /// Fotos de trabalhos anteriores (o "catálogo" do profissional).
  final List<String> catalogo;

  /// Avaliações recebidas. A lista pode crescer quando o cliente avalia.
  final List<Avaliacao> avaliacoes;

  Prestador({
    required this.id,
    required this.nome,
    required this.categoriaId,
    required this.especialidade,
    required this.cidade,
    required this.sobre,
    this.fotoUrl,
    this.verificado = false,
    List<String>? catalogo,
    List<Avaliacao>? avaliacoes,
  })  : catalogo = catalogo ?? [],
        avaliacoes = avaliacoes ?? [];

  /// Média das estrelas (0 se ainda não tem avaliações).
  double get mediaEstrelas {
    if (avaliacoes.isEmpty) return 0;
    final soma = avaliacoes.fold<int>(0, (t, a) => t + a.estrelas);
    return soma / avaliacoes.length;
  }

  /// Quantidade de avaliações recebidas.
  int get totalAvaliacoes => avaliacoes.length;

  /// Primeira letra do nome, usada no avatar quando não há foto.
  String get inicial => nome.isNotEmpty ? nome[0].toUpperCase() : '?';
}

File: android-app/app/src/main/java/com/servicos/app/model/Prestador.kt
package com.servicos.app.model

import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.snapshots.SnapshotStateList

/**
 * Um prestador de serviço cadastrado no app.
 *
 * A lista [avaliacoes] é uma SnapshotStateList: quando um cliente adiciona
 * uma avaliação nova, as telas que mostram esse prestador se atualizam sozinhas.
 */
class Prestador(
    val id: String,
    val nome: String,
    val categoriaId: String,
    val especialidade: String,
    val cidade: String,
    val sobre: String,
    val verificado: Boolean = false,
    val catalogo: List<String> = emptyList(),
    avaliacoesIniciais: List<Avaliacao> = emptyList(),
) {
    val avaliacoes: SnapshotStateList<Avaliacao> =
        mutableStateListOf<Avaliacao>().apply { addAll(avaliacoesIniciais) }

    val totalAvaliacoes: Int
        get() = avaliacoes.size

    /** Média das estrelas (0.0 se ainda não tem avaliações). */
    val mediaEstrelas: Double
        get() = if (avaliacoes.isEmpty()) 0.0
        else avaliacoes.sumOf { it.estrelas }.toDouble() / avaliacoes.size

    /** Primeira letra do nome, usada no avatar quando não há foto. */
    val inicial: String
        get() = nome.firstOrNull()?.uppercase() ?: "?"
}
